client/state/current-user/reducer.js

```diff
@@ -82,9 +90,28 @@ export const capabilities = createReducer( {}, {
 	}
 }, capabilitiesSchema );
 
+export const gravatarStatus = createReducer( {}, {
+	[ GRAVATAR_UPLOAD_START ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: true
+		} );
+	},
+	[ GRAVATAR_UPLOAD_SUCCESS ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+	[ GRAVATAR_UPLOAD_FAILURE ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+}, gravatarSchema );
```

Comment:
> or is there someplace else this "cache flush" can happen?

I think I may have found a neat solution that won't require us going back and removing the local image from the state, or refactoring `<Gravatar />`.

But, first, here's what my line of thinking was so far:

From a larger-picture perspective, the flush can happen in 3 ways:
- Remove the `max-age` from the Gravatar image response, and use `no-cache` to revalidate the cached image each time it's fetched ([as described here](https://jakearchibald.com/2016/caching-best-practices/)). This affects the Gravatar service, and would unnecessarily delay getting this out to our users, so we're going with a local solution.
- Storing the local image and showing it to the user until the the max cache time has elasped (currently 5 minutes) 
- Using a cache-busting URL to get the new Gravatar

At first I thought those last two ways (local image & cache-busting URL) would require the same Calypso workarounds: we'd need to store the the new image (local or cache-busted URL), and the expiration date, and then we'd use those to figure out whether we want to show the temporary image, or the usual Gravatar URL.

Instead, we could store the cache-busting URL directly on `state.users.items.[id].avatar_URL` and we would never have to change it back to the original URL.

I can do this do this in `users/reducer.js`, responding to the `GRAVATAR_UPLOAD_REQUEST_RECEIVE` action.

This works around needing to know whether the user we're dealing with is the current user or not. If it is, then the `avatar_URL` is correctly cache-busted, if not, then we still have the right URL to use.

Two issues I foresee with this approach are:
1. What happens if a user is using a Photon URL (there's a reference to it in `<Gravatar />` and `getResizedImageURL()`)
2. Are we always passing the current user object from the state to Gravatar? I think so, but will need to verify

I'm going to try this out and report back. If there are any issues I overlooked, let me know :)

cc @mdawaffe @gwwar